src/uutf.ml

```diff
@@ -165,7 +165,11 @@ let r_encoding s j l =                  (* guess encoding with max. 3 bytes. *)
 
 (* Decode *)
 
-type src = [ `Channel of in_channel | `String of string | `Manual ]
+type src =
+    [ `Channel of in_channel
+    | `String of string
+	| `Substring of string * int * int
```

Comment:
This is badly indented could you use ocp-indent please, there's a .ocp-indent file at the toplevel of the project.